### DISCUSSION TITLE
WebDav on Linux?

### DISCUSSION THREAD

**Discussion by skatkov** [photoprism/photoprism#2468] [General]
I'm very impressed with a project and moving away all the photos from a filesynching service that I've dependent on for years. I managed to connect all mobile phones with success, but somehow can't get WebDav working on my Manjaro machine.

it seems that manjaro expects, that WebDav urls would start off with "dav://" or "davs://" protocol. But i'm receiving "unable to access location" error in case of davs:// or "http error: permanent redirect" in case of dav://.

Could this be somehow related to the fact, that I'm running this in PikaPods? Did anyone had success using WebDav on Linux?

**Comment by graciousgrey** [comment]
We have  no issues with WebDAV on Linux Mint when using https://example.com/originals/ as url. This also works when PhotoPrism is running on PikaPods. 
Maybe a community member knows more about Manjaro

**Comment by tomaszduda23** [comment]
WebDAV seems to only working with PHOTOPRISM_AUTH_MODE=password

With PHOTOPRISM_AUTH_MODE=public I had the same error as you. In this case response is as follow.

```
curl -i -X PROPFIND http://127.0.0.1:2342/originals/ -H "Depth: 1"
HTTP/1.1 200 OK
Content-Security-Policy: frame-ancestors 'none';
Content-Type: text/html; charset=utf-8
X-Frame-Options: DENY
Date: Sun, 18 Sep 2022 20:45:00 GMT
Transfer-Encoding: chunked

<!DOCTYPE html>
<html lang="en" data-color-mode="dark" data-light-theme="light" data-dark-theme="dark" class="overflow-y-hidden">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
```

**Reply by lastzero** [reply]
Working as described in our documentation: https://docs.photoprism.app/user-guide/sync/webdav/

**Reply by tomaszduda23** [reply]
I just noticed that. Maybe you could consider to change in UI to indicates WebDav inactivity. It was kind of confusing at least for me.

**Reply by lastzero** [reply]
Are the Sync settings visible in the UI when public mode is enabled? What specific UI changes do you suggest?

**Reply by tomaszduda23** [reply]
Sync settings are visible.
I was checking this page. It would be nice if there is information that WebDav does not work.

![Screenshot from 2022-09-18 23-24-19](https://user-images.githubusercontent.com/35012788/190928706-120013b1-b327-4ff4-b2a1-405f1497a85b.png)

**Reply by lastzero** [reply]
How did you get there? AFAIK it isnt linked but I could be wrong.

**Reply by tomaszduda23** [reply]
Sync inside settings or library. I don't remember exactly. Than 'connect via webdav'.

**Reply by tomaszduda23** [reply]
Actually I set PHOTOPRISM_AUTH_MODE=public since my reverse proxy has SSO so I was going to login that way. So maybe it would be better to separate both since there is also PHOTOPRISM_DISABLE_WEBDAV is someone needs it.

**Reply by lastzero** [reply]
We do not yet support proxy forward auth and do not want to be responsible if users unknowingly operate a public warez server, because that is what it would be without further measures.

**Reply by tomaszduda23** [reply]
PHOTOPRISM_AUTH_MODE=public_with_webdav that could be an option.

**Reply by lastzero** [reply]
Proxy Forward Auth is the right option to implement. If we had a larger team, you would have it by now, but 95% of our users do not support our work by contributing or becoming a member.

**Reply by lastzero** [reply]
https://github.com/photoprism/photoprism/commit/cc8bab446df3fb973fe3ede09d0cc0fe4af50f1d hides the Sync tab in Settings if Public Mode is enabled because WebDAV can't be used. I'm not sure why we kept this, but you're right that it caused confusion. Also, configuring connections to servers seems risky on a public instance. You should still be able to upload files in public mode once a connection has been configured.

**Reply by lastzero** [reply]
Our documentation has been updated accordingly: https://github.com/photoprism/photoprism-docs/commit/50c76c62b05c40d7f6da2b1ea90ac586bc503081
